packages/api/migration/1668517874752-AddHUIDataSource.ts

```
@@ -0,0 +1,103 @@
+    );
+
+    await queryRunner.query(`DROP MATERIALIZED VIEW "latest_data"`);
+
+    await queryRunner.query(
+      `CREATE TYPE "public"."metrics_metric_enum_old" AS ENUM('air_temperature', 'barometric_pressure_bottom', 'barometric_pressure_top', 'barometric_pressure_top_diff', 'bottom_temperature', 'cholorophyll_concentration', 'cholorophyll_rfu', 'conductivity', 'dhw', 'odo_concentration', 'odo_saturation', 'ph', 'ph_mv', 'precipitation', 'pressure', 'rh', 'salinity', 'satellite_temperature', 'significant_wave_height', 'sonde_battery_voltage', 'sonde_cable_power_voltage', 'sonde_wiper_position', 'specific_conductance', 'sst_anomaly', 'tds', 'temp_alert', 'temp_weekly_alert', 'top_temperature', 'total_suspended_solids', 'turbidity', 'water_depth', 'wave_mean_direction', 'wave_mean_period', 'wave_peak_period', 'wind_direction', 'wind_gust_speed', 'wind_speed')`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "metrics" ALTER COLUMN "metric" TYPE "public"."metrics_metric_enum_old" USING "metric"::"text"::"public"."metrics_metric_enum_old"`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "time_series" ALTER COLUMN "metric" TYPE "public"."metrics_metric_enum_old" USING "metric"::"text"::"public"."metrics_metric_enum_old"`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "forecast_data" ALTER COLUMN "metric" TYPE "public"."metrics_metric_enum_old" USING "metric"::"text"::"public"."metrics_metric_enum_old"`,
+    );
+    await queryRunner.query(`DROP TYPE "public"."metrics_metric_enum"`);
+    await queryRunner.query(
+      `ALTER TYPE "public"."metrics_metric_enum_old" RENAME TO "metrics_metric_enum"`,
+    );
+
+    await queryRunner.query(
+      `CREATE TYPE "public"."sources_type_enum_old" AS ENUM('gfs', 'hobo', 'metlog', 'noaa', 'sofar_wave_model', 'sonde', 'spotter')`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "forecast_data" ALTER COLUMN "source" DROP DEFAULT`,
```

do we need to make any changes to forecast_data?

---

psql complains altering column `source` without  dropping the default value first. It is later restored:
```
await queryRunner.query(
      `ALTER TABLE "forecast_data" ALTER COLUMN "source" SET DEFAULT 'sofar_wave_model'`,
    );
```

---

Ok, thinking about it a bit more, should it even have a default value? It could be misleading if we forget to set it explicitly and then assign "sofar_wave_model". What do you think?

---

Yes that sounds good. forecast_data are being set only in `hindcast-wind-wave-data.ts` where source is explicitly set. I am removing the default value in this migration.